### PR TITLE
Kiwix desktop jammy appimage

### PR DESCRIPTION
**Pull request by veloman-yunkan**
Fixes kiwix/kiwix-desktop#1219
 
Depends on kiwix/container-images#264

A working appimage was created using the updated `scripts/create_kiwix-desktop_appImage.sh` in a docker container build using the following Dockerfile  `kiwix-build_ci/jammy_builder.dockerfile` introduced by kiwix/container-images#264).

### CODE REVIEW

**Comment by veloman-yunkan** [comment]
So we need to introduce a `jammy` builder. The big question is which of our CI/CD jobs should we switch to that new build platform? Should we only create the `kiwix-desktop` appimage under jammy or some other builds should be upgraded as well?

**Comment by kelson42** [comment]
@veloman-yunkan Great but I don't know how/what to answer to you. We probably need to move the whole linux compilation ci/cd on jammy.

**Comment by veloman-yunkan** [comment]
> @veloman-yunkan Great but I don't know how/what to answer to you. We probably need to move the whole linux compilation ci/cd on jammy.

@kelson42 Then I believe that such a transition shouldn't be done as part of this PR. Rather a separate ticket must be opened and the pros and cons of the proposed upgrade should be discussed there in a broader scope.

**Comment by kelson42** [comment]
@veloman-yunkan So what would be the solution for `2.4.0`?

**Comment by veloman-yunkan** [comment]
> @veloman-yunkan So what would be the solution for `2.4.0`?

Maybe we can release only `kiwix-desktop` (and its appimage) from under jammy?

**Comment by kelson42** [comment]
> > @veloman-yunkan So what would be the solution for `2.4.0`?
> 
> Maybe we can release only `kiwix-desktop` (and its appimage) from under jammy?

As this stage, I believe this could be a temporary solution indeed.

**Comment by veloman-yunkan** [comment]
The PR is mostly ready. The most recent CI run generated an appimage built under jammy (https://tmp.kiwix.org/ci/dev_preview/kiwix-desktop_jammy_appimage/kiwix-desktop_x86_64_2024-11-25.appimage) that is devoid of the issue reported in kiwix/kiwix-desktop#1219.

The failure of the two CI runs under manylinux seems to be because of the change of the builder docker image tag from `2024-06-03` to `latest`.

**Comment by veloman-yunkan** [comment]
FYI, regarding the flatpak version of kiwix-desktop - it (https://tmp.kiwix.org/ci/dev_preview/kiwix-desktop_jammy_appimage/org.kiwix.desktop.2024-11-25.flatpak which was built under focal) works fine since a newer version of Qt is used in Flatpak.

**Comment by veloman-yunkan** [comment]
> The failure of the two CI runs under manylinux seems to be because of the change of the builder docker image tag from `2024-06-03` to `latest`.

I cannot find where the manylinux version of the docker image is hosted. https://ghcr.io/kiwix/kiwix-build_ci_focal works https://ghcr.io/kiwix/kiwix-build_ci_manylinux doesn't.

**Comment by kelson42** [comment]
> The PR is mostly ready. The most recent CI run generated an appimage built under jammy (https://tmp.kiwix.org/ci/dev_preview/kiwix-desktop_jammy_appimage/kiwix-desktop_x86_64_2024-11-25.appimage) that is devoid of the issue reported in [kiwix/kiwix-desktop#1219](https://github.com/kiwix/kiwix-desktop/issues/1219).
> 
> The failure of the two CI runs under manylinux seems to be because of the change of the builder docker image tag from `2024-06-03` to `latest`.

Maybe installing the `requests` Python module? https://stackoverflow.com/questions/17309288/importerror-no-module-named-requests

**Comment by kelson42** [comment]
> > The PR is mostly ready. The most recent CI run generated an appimage built under jammy (https://tmp.kiwix.org/ci/dev_preview/kiwix-desktop_jammy_appimage/kiwix-desktop_x86_64_2024-11-25.appimage) that is devoid of the issue reported in [kiwix/kiwix-desktop#1219](https://github.com/kiwix/kiwix-desktop/issues/1219).
> > The failure of the two CI runs under manylinux seems to be because of the change of the builder docker image tag from `2024-06-03` to `latest`.
> 
> Maybe installing the `requests` Python module? https://stackoverflow.com/questions/17309288/importerror-no-module-named-requests

@benoit74 @rgaudin Do you have an idea why the CI fails? The packages `requests` seems installed properly !!? See https://github.com/kiwix/container-images/blob/main/kiwix-build_ci/manylinux_builder.dockerfile#L16

**Comment by rgaudin** [comment]
@kelson42, again, don't look at the *Upload failure logs* output. The problem is _always_ in previous steps. In this one, it's in the checkout one: `pip3: command not found`

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -131,12 +131,14 @@ jobs:
...
           - config: aarch64_mixed
             image_variant: manylinux
+          - config: native_dyn
+            image_variant: jammy
     env:
       HOME: /home/runner
       SSH_KEY: /tmp/id_rsa
     runs-on: ubuntu-22.04
     container:
-      image: "ghcr.io/kiwix/kiwix-build_ci_${{matrix.image_variant}}:2024-06-03"
```
Comment (kelson42) [inline .github/workflows/cd.yml:139]: Please come back to fixed/pinned dependency

**Review by kelson42 — CHANGES_REQUESTED** [review]
Looks good to me and could be considered ready to merge once:
* CI is fixed, see https://github.com/kiwix/container-images/pull/265
* Docker image dependencies are pinned (again)
* @rgaudin has approved on his side

**Review by rgaudin — APPROVED** [review]
LGTM ; am also in favor of dedicated ticket with pros/cons for the general switch

File: .github/workflows/cd.yml (hunk 1 above)
Comment (veloman-yunkan) [reply .github/workflows/cd.yml:139]: Done

**Pull request merged**